### PR TITLE
Implement access management for creating a test

### PR DESCRIPTION
**Pull request by kstam**

### CODE REVIEW

File: src/application/service/index.ts [hunk 1]
```diff
@@ -110,7 +110,7 @@ export const getTestTypes = new GetTestTypes(testTypeRepository);
 
 export const createSharingCode = new CreateSharingCode(sharingCodeRepository);
 
-export const createOrUpdateTest = new CreateOrUpdateTest(
+export const createOrUpdateTest = new CreateTest(
```
Comment (StevePole) [inline src/application/service/index.ts]: How come you didn't rename this one?

File: src/application/service/CreateTest.ts [hunk 2]
```diff
@@ -2,32 +2,64 @@ import { UserId } from '../../domain/model/user/UserId';
...
-export class CreateOrUpdateTest {
+export class CreateTest {
   constructor(
     private testRepository: TestRepository,
     private testTypeRepository: TestTypeRepository
   ) {}
 
   public async execute(
-    actorUserId: string,
+    actor: User,
```
Comment (StevePole) [inline src/application/service/CreateTest.ts]: Sig is a bit odd, two users but one is passed as an object, one as an id?  I have been a bit confused over which approach is preferred on these.

File: src/application/service/index.ts (hunk 1 above)
Comment (kstam) [reply src/application/service/index.ts:113]: Whooops. missed that in the refactor

File: src/application/service/CreateTest.ts (hunk 2 above)
Comment (kstam) [reply src/application/service/CreateTest.ts:22]: Good point. I initially also kept it with merely passing the actorId but then I realised that the actor will always be present on the presentation and we are querying the user repo twice. So for efficiency's shake I opted to send down/and make the application service require an instance of actor.

**Pull request merged**